test/grouping.jl

```diff
@@ -1339,6 +1339,13 @@ end
         @test gd[NamedTuple(key)] ≅ gd[i]
         # Plain tuple
         @test gd[Tuple(key)] ≅ gd[i]
+        # Dict, with `String` as default
```

Comment:
Please add some tests for `Dict` where `keytype` is `AbstractString` and `SubString` (to check my fix above).